app/views/users/show.html.haml

```diff
@@ -1,9 +1,14 @@
-%br/
+#profile
+  %h2
+    User: #{@user.username}
+  - if user_signed_in? && (current_user == @user || current_user.admin?)
```

Comment:
This is fine, but one slightly more concise way to do it would be:

`if current_user == @user || current_user&.admin?`

The safe navigation operator `&.` is really useful in conjuction with 'current_user' because you can still call methods on it without first checking whether there is a current_user.